app/services/users.js

```diff
@@ -12,18 +15,47 @@ const signUp = user =>
     if (foundUser) {
       throw errors.userExistsError('User already exists');
     }
-    return models.users
-      .create(user)
-      .then(createdUser => {
-        logger.info(`New user created: ${createdUser.email}`);
-        return createdUser;
-      })
-      .catch(error => {
-        throw errors.databaseError(error.message);
-      });
+
+    return hash(user.password).then(hashedPassword => {
+      user.password = hashedPassword;
+      return models.users
+        .create(user)
+        .then(createdUser => {
+          logger.info(`New user created: ${createdUser.email}`);
+          return createdUser;
+        })
+        .catch(error => {
+          throw errors.databaseError(error.message);
+        });
+    });
+  });
+
+const createToken = email => {
+  const token = jwt.sign({ email }, config.common.session.secret, { expiresIn: '24h' });
+  return {
+    success: true,
+    message: 'Authentication successful',
+    token
+  };
+};
+
+const signIn = user =>
```

Comment:
it isn't the responsibility of the service to check if user exists or if the password is correct

Reply:
I am finding it difficult to find where it should go. 
Should these checks go in the middleware for sign up? Or in the controller?

Reply:
Logic was moved to an interactor in latest commit!